extras/ping/ping.go

```diff
@@ -501,7 +508,7 @@ func (p *Pinger) recvICMP(recv chan<- *packet) error {
 			select {
 			case <-p.done:
 				return nil
-			case recv <- &packet{bytes: buf, nbytes: n}:
+			case recv <- &packet{bytes: buf[:], nbytes: n}:
```

Comment:
The `buf[:]` idiom creates a new slice using the same underlying array. If the intent is to copy the array to make sure it is not otherwise modified, I think you want `append([]byte{}, buf...)`.

Please, see [this playground](https://go.dev/play/p/xYLfTJHLGvi) for more information about the difference between the two idioms.

Assuming the intent here is to duplicate the bytes, I would like to better understand why. Because we are not aiming for high high performance, I would say we should just allocate a new buffer for each message and accept that we transfer its ownership when we deliver it over a channel.

(You may have seen the `buf[:]` idiom being used when the backing buffer is a `var buf [256]byte` rather than a `buf := make([]byte. 256)`. The reason why it's necessary to use `buf[:]` in such a case is that you have a function that accepts a `[]byte` - i.e., a slice - and so you need to convert from an array to a slice, and you end up doing that using `buf[:]`.)